gcloud/pubsub/connection.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Create / interact with gcloud pubsub connections."""
+
+import json
+
+from six.moves.urllib.parse import urlencode  # pylint: disable=F0401
+
+from gcloud import connection as base_connection
+from gcloud.exceptions import make_exception
+
+
+class Connection(base_connection.Connection):
```

Comment:
Yeah I think it'll be easiest to reason about / review if you or I move `storage.Connection` to `gcloud.connection.DiscoveryConnection` and then rebase this commit on top of it.

There is lots of duplicated code here.